### PR TITLE
feat(spans): extract cache size and cache hit in indexed spans

### PR DESCRIPTION
**Pull request by DominikB2014**
Work for https://github.com/getsentry/sentry/issues/68118

This PR adds
1. `cache.hit` tag for indexed cached spans (I believe we only support string values here, so i converted to string, please lmk otherwise!)
2. `cache.item_size` measurement for indexed cached spans

#skip-changelog

### CODE REVIEW

File: relay-event-normalization/src/normalize/span/tag_extraction.rs [hunk 1]
```diff
@@ -406,6 +410,18 @@ pub fn extract_tags(
...
+        if span_op.starts_with("cache.") {
+            if let Some(cache_hit) = span.data.value().and_then(|data| data.cache_hit.value()) {
+                match cache_hit {
+                    Value::Bool(true) => span_tags.insert(SpanTagKey::CacheHit, "true".to_owned()),
+                    Value::Bool(false) => {
+                        span_tags.insert(SpanTagKey::CacheHit, "false".to_owned())
+                    }
+                    _ => None,
+                };
+            }
```
Comment (jjbayer) [inline relay-event-normalization/src/normalize/span/tag_extraction.rs]:
```suggestion
            if let Some(Value::Bool(cache_hit)) = span.data.value().and_then(|data| data.cache_hit.value()) {
                let tag_value = if cache_hit { "true" } else { "false" };
                span_tags.insert(SpanTagKey::CacheHit, tag_value.to_owned());
            }
```

File: relay-event-normalization/src/normalize/span/tag_extraction.rs [hunk 2]
```diff
@@ -104,6 +106,8 @@ impl SpanTagKey {
             SpanTagKey::TimeToInitialDisplay => "ttid",
             SpanTagKey::FileExtension => "file_extension",
             SpanTagKey::MainThread => "main_thread",
+            SpanTagKey::CacheHit => "cache.hit",
+            SpanTagKey::CacheItemSize => "cache.item_size",
```
Comment (jjbayer) [inline relay-event-normalization/src/normalize/span/tag_extraction.rs]:
```suggestion
```

File: relay-event-normalization/src/normalize/span/tag_extraction.rs [hunk 3]
```diff
@@ -559,6 +575,27 @@ pub fn extract_measurements(span: &mut Span) {
...
+        if let Some(data) = span.data.value() {
+            if let Some(value) = match &data.cache_item_size.value() {
+                Some(Value::F64(f)) => Some(*f),
+                Some(Value::I64(i)) => Some(*i as f64),
+                Some(Value::U64(u)) => Some(*u as f64),
+                _ => None,
+            } {
+                let measurements = span.measurements.get_or_insert_with(Default::default);
+                measurements.insert(
+                    SpanTagKey::CacheItemSize.sentry_tag_key().into(),
```
Comment (jjbayer) [inline relay-event-normalization/src/normalize/span/tag_extraction.rs]:
```suggestion
                    "cache.item_size".to_owned(),
```

File: relay-event-normalization/src/normalize/span/tag_extraction.rs [hunk 4]
```diff
@@ -1283,6 +1320,83 @@ LIMIT 1
...
+        extract_span_tags_from_event(&mut event, 200);
+
+        let span_1 = &event.spans.value().unwrap()[0];
+        let span_2 = &event.spans.value().unwrap()[1];
+
+        let tags_1 = get_value!(span_1.sentry_tags).unwrap();
+        let tags_2 = get_value!(span_2.sentry_tags).unwrap();
+        let measurements_1 = span_1.value().unwrap().measurements.value().unwrap();
+        println!("MEASUREMENTS!");
+        println!("{:?}", measurements_1);
```
Comment (jjbayer) [inline relay-event-normalization/src/normalize/span/tag_extraction.rs]:
```suggestion
```

**Review by jjbayer — APPROVED** [review]
LGTM, please see comments before merging.

File: relay-event-schema/src/protocol/span.rs [hunk 5]
```diff
@@ -262,10 +262,18 @@ pub struct SpanData {
     #[metastructure(field = "resource.render_blocking_status")]
     pub resource_render_blocking_status: Annotated<Value>,
 
-    /// Name of the database host.
+    /// Name of the web server host.
     #[metastructure(field = "server.address")]
     pub server_address: Annotated<Value>,
 
+    /// Wether cache was hit or miss on a read operation.
```
Comment (iker-barriocanal) [inline relay-event-schema/src/protocol/span.rs]:
```suggestion
    /// Whether cache was hit or miss on a read operation.
```

File: relay-event-schema/src/protocol/span.rs
Comment (iker-barriocanal) [inline relay-event-schema/src/protocol/span.rs:1]: Are there plans to extend the [span data docs](https://develop.sentry.dev/sdk/performance/span-data-conventions/)?

File: relay-event-normalization/src/normalize/span/tag_extraction.rs [hunk 6]
```diff
@@ -1283,6 +1320,83 @@ LIMIT 1
         assert!(!tags_3.contains_key("raw_domain"));
     }
 
+    #[test]
+    fn test_cache_extraction() {
+        let json = r#"
+            {
+                "spans": [
+                {
```
Comment (iker-barriocanal) [inline relay-event-normalization/src/normalize/span/tag_extraction.rs]: nit: is the indentation right?

**Review by iker-barriocanal — APPROVED** [review]

File: relay-event-schema/src/protocol/span.rs
Comment (DominikB2014) [reply relay-event-schema/src/protocol/span.rs:1]: Hmm, could you clarify what you expect to be added there? Both cache.hit and cache.item_size are documented there

Comment (iker-barriocanal) [reply relay-event-schema/src/protocol/span.rs:1]: Oops, must have looked at a cached page. Thanks for documenting it!

File: relay-event-normalization/src/normalize/span/tag_extraction.rs (hunk 6 above)
Comment (DominikB2014) [reply relay-event-normalization/src/normalize/span/tag_extraction.rs]: updated!

File: relay-event-schema/src/protocol/span.rs [hunk 7]
```diff
@@ -231,10 +231,18 @@ pub struct SpanData {
     #[metastructure(field = "resource.render_blocking_status")]
     pub resource_render_blocking_status: Annotated<Value>,
 
-    /// Name of the database host.
+    /// Name of the web server host.
```
Comment (DominikB2014) [inline relay-event-schema/src/protocol/span.rs:234]: This was just an incorrect comment, I updated to reflect the value better

**Pull request merged**